mindsdb/integrations/libs/llm_utils.py

```diff
@@ -0,0 +1,41 @@
+import re
+import numpy as np
+
+
+def get_completed_prompts(base_template, df):
+    """
+        Helper method that produces formatted prompts given a template and data in a Pandas DataFrame.
+        It also returns the ID of any empty templates that failed to be filled due to missing data.
+    """
+    columns = []
+    spans = []
+    matches = list(re.finditer("{{(.*?)}}", base_template))
+
+    assert len(matches) > 0, 'No placeholders found in the prompt, please provide a valid prompt template.'
```

Comment:
Do we need to raise exception here?

Reply:
Yes, otherwise it is a static prompt for every column, and data passing becomes meaningless 😛